pkg/instancetype/instancetype.go

```diff
@@ -333,6 +351,66 @@ func getInstancetypeAPISpec(obj runtime.Object) (interface{}, error) {
 	}
 }
 
+func checkCPUPreferenceRequirements(instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) (*k8sfield.Path, error) {
+	if instancetypeSpec != nil {
+		if instancetypeSpec.CPU.Guest < preferenceSpec.Requirements.CPU.Guest {
+			return k8sfield.NewPath("spec", "instancetype"), fmt.Errorf(InsufficientInstanceTypeCPUResourcesErrorFmt, instancetypeSpec.CPU.Guest, preferenceSpec.Requirements.CPU.Guest)
+		}
+	} else {
+		if vmiSpec.Domain.CPU == nil {
+			return k8sfield.NewPath("spec", "template", "spec", "domain", "cpu"), fmt.Errorf(NoVMCPUResourcesDefinedErrorFmt, preferenceSpec.Requirements.CPU.Guest)
```

Comment:
You can extract the field. At least define something like `cpuField := k8sfield.NewPath("spec", "template", "spec", "domain", "cpu") `.